### PR TITLE
Async 🎉

### PR DESCRIPTION
**Pull request by marcaurele**

### CODE REVIEW

**Comment by vincentbernat** [comment]
You should put that in a separate module. No need of `if has_async` and compatibility with Python 2.7 preserved.

File: README.rst [hunk 1]
```diff
@@ -8,6 +8,7 @@ CS
 A simple, yet powerful CloudStack API client for python and the command-line.
 
 * Python 2.6+ and 3.3+ support.
+* Async support for 3.5+
```
Comment (brutasse) [inline README.rst]: "Python 3.5+"

File: README.rst [hunk 2]
```diff
@@ -133,6 +138,61 @@ Or in Python::
...
+    cs = AIOCloudStack(**read_config(), job_timeout=300, poll_interval=5)
+
+Async deployment of multiple vms
+________________________________
+
+.. code-block:: python
+
+    import asyncio
+    from cs import AIOCloudStack, read_config
+    acs = AIOCloudStack(**read_config('preprod:testing'))
```
Comment (brutasse) [inline README.rst]: `**read_config()`

Not sure it's worth calling the async client differently (first async example uses `cs`)

File: README.rst [hunk 3]
```diff
@@ -133,6 +138,61 @@ Or in Python::
...
+            if t.exception():
+                exceptions += 1
+                last_exception = t.exception()
+            elif t.result():
+                results.append(t.result())
+        if exceptions:
+            print(f"{exceptions} deployment(s) failed")
+            raise last_exception
+    except Exception:
+        raise
```
Comment (brutasse) [inline README.rst]: try… except doesn't do anything

File: README.rst [hunk 4]
```diff
@@ -133,6 +138,61 @@ Or in Python::
...
+
+    import asyncio
+    from cs import AIOCloudStack, read_config
+    acs = AIOCloudStack(**read_config('preprod:testing'))
+    tasks = [asyncio.ensure_future(acs.deployVirtualMachine(zoneid='',
+                                                            serviceofferingid='',
+                                                            templateid='')) for _ in range(5)]
+    results = []
+    try:
+        done_t, pending_t = await asyncio.wait(tasks)
```
Comment (brutasse) [inline README.rst]: `done, pending` is more consitent with asyncio docs

File: cs/__init__.py [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    from ConfigParser import NoSectionError
+
+try:
+    import pygments
+    from pygments.lexers import JsonLexer
+    from pygments.formatters import TerminalFormatter
+except ImportError:
+    pygments = None
+
+from .client import read_config, CloudStack, CloudStackException  # noqa
```
Comment (marcaurele) [inline cs/__init__.py]: .client is missing

Comment (brutasse) [reply cs/__init__.py]: Yep, added :)

**Comment by marcaurele** [comment]
:shipit:

**Pull request merged**